src/sdk/namespace_fs.js

```diff
@@ -513,7 +549,37 @@ class NamespaceFS {
                 if (r.common_prefix) {
                     res.common_prefixes.push(r.key);
                 } else {
-                    res.objects.push(this._get_object_info(bucket, r.key, r.stat, 'null'));
+                    const obj_info_list = [];
+                    const obj_info = this._get_object_info(bucket, r.key, r.stat, 'null', true);
+                    if (!list_versions && obj_info.delete_marker) {
+                        continue;
+                    }
+                    if (obj_info.delete_marker) {
```

Comment:
delete marker can be latest version if there is no current version
for example:
put object (version1 created)
delete object (delete marker was created and latest)

Reply:
If the object is deleted then do we have to keep it `latest` ? 🤔

Reply:
this can be checked on AWS :) but I guess that the answer is yes